springUsers/src/main/java/com/example/springUsers/Services/Implementations/UserServiceImplementation.java

```
@@ -0,0 +1,106 @@
+package com.example.springUsers.Services.Implementations;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.springframework.beans.BeanUtils;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageRequest;
+import org.springframework.stereotype.Service;
+
+import com.example.springUsers.Dao.UserRepository;
+import com.example.springUsers.Dto.UserDto;
+import com.example.springUsers.Model.User;
+import com.example.springUsers.Services.UserServices;
+import com.example.springUsers.Shared.Utils;
+
+@Service
+public class UserServiceImplementation implements UserServices {
+
+	private UserRepository userRepository;
+	private Utils utils; 
+
+	public UserServiceImplementation(UserRepository userRepository, Utils utils) {
+		this.userRepository = userRepository;
+		this.utils = utils; 
+	}
+
+	@Override
+	public List<UserDto> getAllUsers(int page, int limit) {
+		
+		List<User> userList = new ArrayList<User>(); 
+		
+		if (page>0) page --; 
+		PageRequest pageableRequest = PageRequest.of(page, limit); 
+		
+		Page<User> userPageList = userRepository.findAll(pageableRequest);
+		
+		userList = userPageList.getContent(); 
+		
+		
+		List<UserDto> userDtoList = new ArrayList<UserDto>(); 
+		
+		for ( int i = 0; i<userList.size(); i++ ) {
+			UserDto userDto = new UserDto(); 
+			BeanUtils.copyProperties(userList.get(i), userDto);
+			userDtoList.add(userDto); 
+		}
+		
+		
+		return userDtoList;
+	}
+
+	@Override
+	public UserDto getUserByUserId(String userId) {
+		User user = userRepository.findByUserId(userId); 
```

if you don't find the user you can throw an exception right from here